src/components/map/Mapbox.vue

```diff
@@ -48,21 +63,35 @@ export default {
         center: [-123.120735, 49.28273],
         zoom: 12
       },
+      geocoderOptions: {
+        placeholder: "Search events and locations"
+      },
+
+      searchQuery: "",
+      suggestions: []
     };
   },
 
   created() {
     this.map = null;
+    this.geocoderControl = null;
   },
+  mounted() {},
 
   methods: {
     onMapLoaded(event) {
-      // console.log("Map loaded!");
       this.map = event.map;
     },
     navigateToEvent(event) {
       const { _id: id } = event;
       this.$router.push({ path: `/eventpage/${id}` });
+    },
+    handleSearch() {
+      // TODO: middle marker, and possible backend stuff for 'user suggestions'
```

Comment:
actually i don't really like middle markers i feel like the map is cleaner without it

Reply:
It seems like industry standard - even Google Maps has it. Is there a better way you think we can implement to 'show' where exactly an event/location is, then?

Reply:
I know but isn't the point of us using mapbox so it's cleaner and more aesthetic like snap map? Also I feel like the text is dynamic enough and if they were searching for an event there already is a marker.

Reply:
Makes total sense for event searching! Personally it would be cool to have some kind of animation play for event searching: [inspo](https://i.stack.imgur.com/lfmSi.gif).

For location searching its an entirely different case, since there aren't markers already there.